File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
      int n=nums.length,j=0,k=0;
      int[] pos=new int[n/2];
      int[] neg= new int[n/2];
      for(int i=0;i<n;i++){
        if(nums[i]>0){
            pos[j++]=nums[i];
        }
        else neg[k++]=nums[i];
      }
      j=0;k=0;
      for(int i=0;i<n;i=i+2){
        nums[i]=pos[j++];
        nums[i+1]=neg[k++];
      }
      return nums;
    }
}

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int n=nums.size();
        vector<int> pos(n/2); vector<int> neg(n/2); int p=0, m=0;
        for(int i=0;i<n;i++){
            if(nums[i]>0) pos[p++]=nums[i];
            else{
                neg[m++]=nums[i];
            }
        }
        p=0; m=0;
        for(int i=0;i<n;i++){
            if(i%2==0) nums[i]=pos[p++];
            else nums[i]=neg[m++];
        }
        return nums;
    }
};